src/cpp/rtps/builtin/discovery/endpoint/EDPUtils.hpp

```diff
@@ -68,7 +68,6 @@ class EDPUtils
         {
             PoolConfig pool_cfg = PoolConfig::from_history_attributes(history_attr);
             pool->release_history(pool_cfg, is_reader);
-            TopicPayloadPoolRegistry::release(pool);
```

Comment:
Shouldn't we do a `pool.reset()` call here?

Reply:
Certainly!